sources/common/src/XmlSignalFactory.cpp

```diff
@@ -0,0 +1,175 @@
+        }
+
+        if (!std::strcmp("name", tag.name()))
+        {
+            m_Name = tag.child_value();
+        }
+
+        if (!std::strcmp("value", tag.name()))
+        {
+            m_Value = tag.child_value();
+        }
+
+        if (!std::strcmp("size", tag.name()))
+        {
+            m_Size = tag.child_value();
+        }
+    }
+}
+
+const std::string & PARSER::XmlDescription::GetRegex() const
+{
+    if (m_RegexLine.length())
+    {
+        return m_RegexLine;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_REGEX_EXPRESSION,
+        "XML - No parsing regular expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetTimestamp() const
+{
+    if (m_Timestamp.length())
+    {
+        return m_Timestamp;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_TIMESTAMP_EXPRESSION,
+        "XML - No timestamp expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetName() const
+{
+    if (m_Name.length())
+    {
+        return m_Name;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_NAME_EXPRESSION,
+        "XML - No name expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetValue() const
+{
+    if (m_Value.length())
+    {
+        return m_Value;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_VALUE_EXPRESSION,
+        "XML - No value expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetSize() const
+{
+    if (m_Size.length())
+    {
+        return m_Size;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_SIZE_EXPRESSION,
+        "XML - No size expression.");
+}
+
+PARSER::XmlSignalFactory::XmlSignalFactory(const std::string& rXmlFileName) : SignalFactory(false)
+{
+    // Parse the XML and add signal creators.
+    // pugi test
+    pugi::xml_document doc;
+    if (!doc.load_file(rXmlFileName.c_str())) throw (1);  /// @todo Fix filename and error
+
+    pugi::xml_node signals = doc.child("signals");
+
+    // Traverse the XML.
+    for (pugi::xml_node signal : signals.children())
+    {
+        if (!std::strcmp("vector", signal.name()))
+        {
+            std::unique_ptr<XmlDescription> description = std::make_unique<XmlDescription>(signal);
+            m_vpSignalCreators.push_back(std::make_unique<XmlISignalCreator>(
+                description->GetRegex(),
+                description->GetTimestamp(),
+                description->GetName(),
+                description->GetValue(),
+                description->GetSize()));
+            continue;
+        }
+
+        if (!std::strcmp("real", signal.name()))
+        {
+            std::unique_ptr<XmlDescription> description = std::make_unique<XmlDescription>(signal);
+            m_vpSignalCreators.push_back(std::make_unique<XmlFSignalCreator>(
+                description->GetRegex(),
+                description->GetTimestamp(),
+                description->GetName(),
+                description->GetValue(),
+                ""));
```

Comment:
Why is this empty? If value is never used why it's even a param? It should not be a constructor param and this default value should be passed to base class constructor in `XmlFSignalCreator` class constructor.